packages/types/src/Type.ts

```diff
@@ -172,12 +172,16 @@ export default class Type extends Text {
   private static _removeTraits (): Mapper {
     return (value: string): string => {
       return value
+        // `T :: AccountId` -> `T::AccountId` https://github.com/paritytech/substrate/issues/1244
+        .replace(/\s+::\s+/g, '::')
         // anything `T::<type>` to end up as `<type>`
         .replace(/T::/g, '')
         // `system::` with `` - basically we find `<T as system::Trait>`
         .replace(/system::/g, '')
         // replace `<T as Trait>::` (possibly sanitised just above)
         .replace(/<T as Trait>::/g, '')
+        // `<TasTrait>::type` -> `type` https://github.com/paritytech/substrate/issues/1244
+        .replace(/<TasTrait>::/g, '')
```

Comment:
Could have possibly gone for the same `\s+` hackery as above with `T::` here

Reply:
Prefer to leave it as 2 lines, the 1st one is necessary, the 2nd is a bug fix